src/dp_abstraction/aca_dataplane_ovs.cpp

```diff
@@ -357,4 +361,345 @@ int ACA_Dataplane_OVS::update_port_state_workitem(const PortState current_PortSt
   return overall_rc;
 }
 
+int ACA_Dataplane_OVS::update_neighbor_state_workitem(NeighborState current_NeighborState,
+                                                      GoalState &parsed_struct,
+                                                      GoalStateOperationReply &gsOperationReply)
+{
+  int rc, overall_rc;
+  struct sockaddr_in sa;
+  string virtual_ip_address;
+  string virtual_mac_address;
+  string host_ip_address;
+  NetworkType found_network_type;
+  uint found_tunnel_id;
+  string found_gateway_mac;
+  string neighbor_host_dvr_mac;
+  bool subnet_info_found = false;
+  ulong culminative_dataplane_programming_time = 0;
+  ulong culminative_network_configuration_time = 0;
+
+  auto operation_start = chrono::steady_clock::now();
+
+  NeighborConfiguration current_NeighborConfiguration =
+          current_NeighborState.configuration();
+
+  assert(current_NeighborConfiguration.format_version() > 0);
+
+  // TODO: need to design the usage of current_NeighborConfiguration.revision_number()
+  assert(current_NeighborConfiguration.revision_number() > 0);
+
+  switch (current_NeighborState.operation_type()) {
+  case OperationType::CREATE:
+
+    if (current_NeighborConfiguration.neighbor_type() == NeighborType::L2 ||
+        current_NeighborConfiguration.neighbor_type() == NeighborType::L3) {
+      try {
+        assert(current_NeighborConfiguration.fixed_ips_size() == 1);
+        virtual_ip_address = current_NeighborConfiguration.fixed_ips(0).ip_address();
+
+        // inet_pton returns 1 for success 0 for failure
+        if (inet_pton(AF_INET, virtual_ip_address.c_str(), &(sa.sin_addr)) != 1) {
+          throw std::invalid_argument("Virtual ip address is not in the expect format");
+        }
+
```

Comment:
If there is only 1 option in the switch, why not just use if...?

Reply:
because we will have more options later :) namely OperationType::DELETE, OperationType::UPDATE